packages/search-input/src/SearchInput/SearchInput.story.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable no-console */
+import React, { ChangeEventHandler, useState } from 'react';
+import { ComponentMeta, ComponentStory } from '@storybook/react';
+import { startCase } from 'lodash';
+
+import { css } from '@leafygreen-ui/emotion';
+import {
+  storybookArgTypes,
+  // @ts-ignore
+  storybookExcludedControlParams,
+} from '@leafygreen-ui/lib';
+
+import { SearchInput, SearchResult, SearchResultGroup } from '..';
+
+import { State } from './SearchInput.types';
+
+export default {
+  title: 'Components/SearchInput',
+  component: SearchInput,
+  argTypes: {
+    darkMode: storybookArgTypes.darkMode,
+    state: { control: 'select', options: Object.values(State) },
+    id: { control: 'string' },
+    placeholder: { control: 'string' },
+    value: { control: 'string' },
+    disabled: { control: 'boolean' },
+    ref: { control: 'none' },
+  },
+  parameters: {
+    controls: {
+      exclude: [
+        ...storybookExcludedControlParams,
+        'value',
+        'id',
+        'showWedge',
+        'onSubmit',
+      ],
+    },
+  },
+} as ComponentMeta<typeof SearchInput>;
+
+export const Basic: ComponentStory<typeof SearchInput> = props => (
+  <SearchInput
+    className={css`
+      width: 200px;
+    `}
+    {...props}
+  />
+);
+
+export const WithResults: ComponentStory<typeof SearchInput> = props => (
+  <SearchInput
+    className={css`
+      width: 200px;
+    `}
+    onChange={() => {
+      console.log('SB: Change');
+    }}
+    {...props}
+  >
+    <SearchResult
+      onClick={() => {
+        console.log('SB: Click Apple');
+      }}
+      description="This is a description"
+    >
+      Apple
+    </SearchResult>
+    <SearchResult>Banana</SearchResult>
+    <SearchResult as="a" href="#" description="This is a link">
+      Carrot
+    </SearchResult>
+    <SearchResult description="This is a very very long description. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor.">
```

Comment:
This might be a nice opportunity to do what I demoed with the new Table component - having a separate story file to allow the user to play with the subcomponent's props using knobs

Reply:
`SearchResults/SearchResults.story.ts`